### ISSUE TITLE
QA for 4/17/2024 production deploy

### ISSUE THREAD

**Issue by ccostino** [GSA/notifications-admin#1429]
### Log In

- [x] notify-staging.app.cloud.gov resolves
- [x] Clicking sign in takes you to login page
- [x] Signing in with Login.gov works as expected
- [x] After logging in, you are brought to the “choose service” screen
- [x] Choosing a service brings you to the dashboard for that service

### Create a template and send

- [x] Clicking “Send messages” brings you to templates page
- [x] When “Create a template” button is clicked, the option to create new or copy existing template appears
- [x] Uploading a csv is quick and automatically directs you to the preview/send page
- [x] Formatting looks clean and in place
- [x] Message dashboard populates and shows “pending” until call is made to AWS for delivery receipt

### Send a message from copied template

- [x] When “Create a template” button is clicked, the option to create new or copy existing template appears
- [x] Clicking “copy an existing template” allows you to choose from pre-existing template
- [x] After editing, the new template saves and you prepare to send

### Send a message from existing template

- [x] Clicking on existing template brings you to preview screen with `prepare to send` button
- [x] All else above persists as true (formatting, send, dashboard, etc)

### Download the csv delivery report

- [x] Clicking on the csv download directly after a message send downloads a report that contains the data we expect
- [x] The batch report appears on the main dashboard batch table with accurate information
- [x] Clicking on the csv download link from the main dashboard downloads a report that contains the data we expect

### Invite a team member

- [x] Clicking on “Team” in settings  brings you to team member screen
- [x] Clicking “Invite a team member” brings you to email and permissions screen
- [x] Clicking invite team member sends an invite and shows confirmation
- [x] Invite email provides live link that works to register new team member

### Delete a team member

- [ ] From team member screen, clicking “Change Details” on a specific profile brings you to that person’s profile
- [ ] Clicking on “Remove this team member” prompts a confirmation
- [ ] Clicking the confirmation removes the team member

### Sign Out

- [x] Hitting “Sign Out” takes you back to the homepage and logs you out

## Specific Issues to review

We have a handful of issues that are ready to be reviewed or could use verification whether or not they're fixed or need more follow up:

- [ ] https://github.com/GSA/notifications-api/issues/645
- [ ] https://github.com/GSA/notifications-admin/issues/1125
- [ ] https://github.com/GSA/notifications-admin/issues/1338
- [ ] https://github.com/GSA/notifications-admin/issues/952
- [ ] https://github.com/GSA/notifications-admin/issues/1270
- [x] https://github.com/GSA/notifications-admin/issues/1269
- [x] https://github.com/GSA/notifications-admin/issues/1427
- [ ] https://github.com/GSA/notifications-api/issues/798
- [ ] https://github.com/GSA/notifications-api/issues/903
- [x] https://github.com/GSA/notifications-admin/issues/1350

## TODO

We're waiting on a couple of things to still be completed and merged to staging:

- [x] Changes to the user settings page to remove sign-in method option
- [x] Changes to the user profile page to remove the option to change password and email (but still show the email address)
- [ ] Changes to the "Set up your profile" template to not mention sending a text code
- [x] Adding the Login.gov button to the "You need to sign in again" template

**Comment by ccostino** [comment]
I think we've already captured this, but noting here again that when you go to create a new template and copy from an existing template, you don't see a list of any of the existing templates and can only search by name, which means you have to remember which templates exist.

**Comment by heyitsmebev** [comment]
If I'm signed out and I somehow land on "You need to sign in again" page...There's no sign in inputs. 

<img width="1153" alt="image" src="https://github.com/GSA/notifications-admin/assets/53159604/f83ae632-ca83-4617-8330-f25564b429cb">

**Comment by ccostino** [comment]
Awesome, thanks for capturing these @heyitsmebev!  We'll follow up to see if we're dealing with known issues or need to create new ones, and if any of these are actual show-stoppers.

As for the sign-in again form there, [that's a known issue](https://github.com/GSA/notifications-admin/issues/1423#issuecomment-2059524818) that @em-herrick has already flagged that still needs to be taken care of - it's on our list of TODOs above! 🙂

**Comment by ccostino** [comment]
Something else that we might need to account for:  when I invited myself to another organization (the TTS organization), I got the email as expected and then clicked on the link.

From there, it prompted me to sign in with Login.gov, which is what I believe we want now, except in this case I was already signed in to the site...

I still signed back in, and was then taken to the set up my profile page, where it said a security code would be texted to me:

<img width="1233" alt="Screenshot 2024-04-16 at 6 11 33 PM" src="https://github.com/GSA/notifications-admin/assets/121565/c0c3e807-c4b3-47ef-9f93-0a29ff89692c">

I filled things in anyway, and then took me to the usage page for the org:

<img width="1259" alt="Screenshot 2024-04-16 at 6 12 59 PM" src="https://github.com/GSA/notifications-admin/assets/121565/121e10f8-31d4-48c4-9a5e-be8795e22872">

And finally I clicked on Team Members, where I see me listed as an invited user with the option to still cancel, even though I had accepted:

<img width="1287" alt="Screenshot 2024-04-16 at 6 12 18 PM" src="https://github.com/GSA/notifications-admin/assets/121565/8156133f-3b32-467d-be19-3dc2864d4bae">

It's entirely possible I broke the flow here by being a platform admin and signed in, inviting myself to an org with the exact same email and going from there.

**Comment by em-herrick** [comment]
A small fix here is that on the `Set up your profile page` under the `Mobile number` the copy here is out-of date, we no longer send security codes.

We should replace with: `We need your number so you can send yourself test texts`

**Comment by ccostino** [comment]
Ah yes, this is what I was seeing yesterday too in the set up your profile form, thanks @em-herrick!

**Comment by A-Shumway42** [comment]
@ccostino The allowance updates to `250,000` when switching the service to live. The values aren't updating now? I sent a few text to verify.

<img width="739" alt="Screen Shot 2024-04-17 at 8 01 41 AM" src="https://github.com/GSA/notifications-admin/assets/90117200/087dcfa6-c4e6-40a4-8ef6-118fa92427dc">

**Comment by jskinne3** [comment]
While investigating #1427 I encountered two broken links:
![Screenshot 2024-04-17 at 9 35 20 AM](https://github.com/GSA/notifications-admin/assets/546123/ca5efef8-fb6c-45a7-bfb9-b294c2bfb729)

Both lead to a "Page not found" at these two URLs:
* https://notify-staging.app.cloud.gov/services//service-settings
* https://notify-staging.app.cloud.gov/services//users

The double slash (`//`) in the path looks like something is missing.

As a result of this I can't say for sure if user story 1427 is complete or not

**Comment by ccostino** [comment]
[Added an issue for the "you need to sign back in" template](https://github.com/GSA/notifications-admin/issues/1437) - the Login.gov button is missing.

**Comment by A-Shumway42** [comment]
@ccostino this was a new service in staging. I logged back into staging and it appears it is working now. I'm not sure what the delay was on the count.
<img width="865" alt="Screen Shot 2024-04-17 at 10 56 52 AM" src="https://github.com/GSA/notifications-admin/assets/90117200/968e507d-2551-425e-9a06-4ca108c81234">

**Comment by ccostino** [comment]
Oh strange, @A-Shumway42!  Two things come to mind that we could look into further to see:

- Was it just because of cached data with your user session - doe they update properly now if you send more messages?
- Is it waiting for the delivery status to update to be successful (and/or failed)?

**Comment by A-Shumway42** [comment]
@ccostino Hard to verify now. I'm not receiving messages to my phone and I've sent a few more in staging. The dashboard still has a count of 2 in the count at the bottom and the top. I also can't access logs in staging because my cf cli can't login...oof.

**Comment by ccostino** [comment]
@A-Shumway42 okay - I noticed some delays with messages going out yesterday myself - you might be able to see some activity (or not!) in AWS in the staging account, but I think we're okay for now and we can follow up on this more if need be.

@heyitsmebev, thanks for all of your reports there!  It looks like the discrepancies with the templates are more so with the QA checklist and not the site itself, so we'll have to update that issue template. 🙂 As for the sign in again page, [that's tracked here and nearly wrapped up!](https://github.com/GSA/notifications-admin/issues/1437)

@jskinne3, thanks for reporting these!  Sure enough those are bugs caused by the current information architecture of all things.  Those two links are tied to a service and only work if you've clicked into a service first - they are settings page for a service, not global/universal for just your user account.  So thank you for flagging, @em-herrick and I confirmed this just a bit ago!  As for the template, thanks for flagging that too; it looks like it didn't get fully updated for some reason, and highlights that the template updating process is still a bit wonky.

@em-herrick and I will write up separate issues to track the things that we do need to follow up on from all of this.  Thank you all!
